### DISCUSSION TITLE
HPIPM interface definitions

### DISCUSSION THREAD

**Discussion by sashankmodali** [casadi/casadi#3206] [General]
Hello everyone,

I am using Casadi to solve an optimal control problem with the nlp solver sqpmethod and qp solver HPIPM. My system has 11 states, 3 inputs, and 1 slack variable, and I have reformulated the problem as having 12 states and 4 inputs. My control horizon is 5. When I try to input the following as options to the solver, I get an error posted below.

I have tried the following options :

1)
N : 5
nx : [0, 12, 12, 12, 12, 12]
nu : [4, 4, 4, 4 ,4 ,0]
ng = [0, 4, 4, 4, 4, 4]

2)
N: 5
nx : [12, 12, 12, 12, 12, 12]
nu : [4, 4, 4, 4 ,4]
ng = [0, 4, 4, 4, 4, 4]

I get the following error for (1):

![image](https://github.com/casadi/casadi/assets/25561571/ffed799c-c0eb-451a-a551-3ee7a2843fb3)

Could anyone provide a clearer definition of how the HPIPM interface works?

Thanks in advance! :)

**Comment by jgillis** [comment]
Best is to look at the relevant examples under test/python/conic.py

**Reply by sashankmodali** [reply]
Thanks a lot for your response! :)

**Comment by jgillis** [comment]
Note that a follow up discussion takes the question further: https://github.com/casadi/casadi/discussions/3277
